src/renderer/features/governance/aggregates/unlockForm.ts

```diff
@@ -0,0 +1,524 @@
+
+const $isMultisig = createStore<boolean>(false);
+const $isProxy = createStore<boolean>(false);
+const $selectedSignatories = createStore<Account[]>([]);
+const $accounts = createStore<Accounts[]>([]);
+
+const $fee = restore(feeChanged, ZERO_BALANCE);
+const $totalFee = restore(totalFeeChanged, ZERO_BALANCE);
+const $multisigDeposit = restore(multisigDepositChanged, ZERO_BALANCE);
+const $isFeeLoading = restore(isFeeLoadingChanged, true);
+
+const $unlockForm = createForm<FormParams>({
+  fields: {
+    shards: {
+      init: [] satisfies AccountWithClaim[],
+      rules: [
+        {
+          name: 'noProxyFee',
+          source: combine({
+            fee: $fee,
+            isProxy: $isProxy,
+            proxyBalance: $proxyBalance,
+          }),
+          validator: (_s, _f, { isProxy, proxyBalance, fee }) => {
+            if (!isProxy) return true;
+
+            return new BN(fee).lte(new BN(proxyBalance));
+          },
+        },
+      ],
+    },
+    signatory: {
+      init: {} as Account,
+      rules: [
+        {
+          name: 'noSignatorySelected',
+          errorText: 'transfer.noSignatoryError',
+          source: $isMultisig,
+          validator: (signatory, _, isMultisig) => {
+            if (!isMultisig) return true;
+
+            return Object.keys(signatory).length > 0;
+          },
+        },
+        {
+          name: 'notEnoughTokens',
+          errorText: 'proxy.addProxy.notEnoughMultisigTokens',
+          source: combine({
+            fee: $fee,
+            isMultisig: $isMultisig,
+            multisigDeposit: $multisigDeposit,
+            signatoryBalance: $signatoryBalance,
+          }),
+          validator: (_s, _f, { fee, isMultisig, signatoryBalance, multisigDeposit }) => {
+            if (!isMultisig) return true;
+
+            return new BN(multisigDeposit).add(new BN(fee)).lte(new BN(signatoryBalance));
+          },
+        },
+      ],
+    },
+    amount: {
+      init: '',
+      rules: [
+        {
+          name: 'required',
+          errorText: 'transfer.requiredAmountError',
+          validator: Boolean,
+        },
+        {
+          name: 'notZero',
+          errorText: 'transfer.notZeroAmountError',
+          validator: (value) => value !== ZERO_BALANCE,
+        },
+        {
+          name: 'insufficientBalanceForFee',
+          errorText: 'transfer.notEnoughBalanceForFeeError',
+          source: combine({
+            fee: $fee,
+            isMultisig: $isMultisig,
+            accounts: $accounts,
+          }),
+          validator: (value, form, { fee, isMultisig, accounts }) => {
+            if (isMultisig) return true;
+
+            return form.shards.every((_: AccountWithClaim, index: number) => {
+              return new BN(fee).lte(new BN(accounts[index]?.balance));
+            });
+          },
+        },
+      ],
+    },
+    description: {
+      init: '',
+      rules: [UnlockRules.description.maxLength],
```

Comment:
you can use more rules from `UnlockRules`, best way if all rules will be stored only in one place